runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/processing/StreamingWorkSchedulerFactory.java

```diff
@@ -0,0 +1,87 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.dataflow.worker.windmill.work.processing;
+
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.apache.beam.runners.dataflow.options.DataflowWorkerHarnessOptions;
+import org.apache.beam.runners.dataflow.worker.DataflowExecutionStateSampler;
+import org.apache.beam.runners.dataflow.worker.DataflowMapTaskExecutorFactory;
+import org.apache.beam.runners.dataflow.worker.HotKeyLogger;
+import org.apache.beam.runners.dataflow.worker.ReaderCache;
+import org.apache.beam.runners.dataflow.worker.streaming.StageInfo;
+import org.apache.beam.runners.dataflow.worker.streaming.harness.StreamingCounters;
+import org.apache.beam.runners.dataflow.worker.streaming.sideinput.SideInputStateFetcher;
+import org.apache.beam.runners.dataflow.worker.util.BoundedQueueExecutor;
+import org.apache.beam.runners.dataflow.worker.windmill.Windmill;
+import org.apache.beam.runners.dataflow.worker.windmill.state.WindmillStateCache;
+import org.apache.beam.runners.dataflow.worker.windmill.work.processing.failures.FailureTracker;
+import org.apache.beam.runners.dataflow.worker.windmill.work.processing.failures.WorkFailureProcessor;
+import org.apache.beam.sdk.fn.IdGenerator;
+import org.joda.time.Instant;
+
+public final class StreamingWorkSchedulerFactory {
+  private StreamingWorkSchedulerFactory() {}
+
+  public static StreamingWorkScheduler createStreamingWorkScheduler(
+      DataflowWorkerHarnessOptions options,
+      Supplier<Instant> clock,
+      ReaderCache readerCache,
+      DataflowMapTaskExecutorFactory mapTaskExecutorFactory,
+      BoundedQueueExecutor workExecutor,
+      Function<String, WindmillStateCache.ForComputation> stateCacheFactory,
+      Function<Windmill.GlobalDataRequest, Windmill.GlobalData> fetchGlobalDataFn,
+      FailureTracker failureTracker,
+      WorkFailureProcessor workFailureProcessor,
+      StreamingCounters streamingCounters,
+      HotKeyLogger hotKeyLogger,
+      DataflowExecutionStateSampler sampler,
+      AtomicInteger maxWorkItemCommitBytes,
+      IdGenerator idGenerator,
+      ConcurrentMap<String, StageInfo> stageInfoMap) {
+    StreamingCommitFinalizer streamingCommitFinalizer =
+        StreamingCommitFinalizer.create(workExecutor);
+    ExecutionStateFactory executionStateFactory =
+        new ExecutionStateFactory(
+            options,
+            mapTaskExecutorFactory,
+            readerCache,
+            stateCacheFactory,
+            sampler,
+            streamingCounters.pendingDeltaCounters(),
+            idGenerator);
+    SideInputStateFetcher sideInputStateFetcher =
+        new SideInputStateFetcher(fetchGlobalDataFn, options);
+
+    return new StreamingWorkScheduler(
```

Comment:
Do we need the factory? How will this be different for the direct path? IT seems that the committer, reader fetch fn etc are injected elsewhere.

If not just have a create method for the streamingworkscheduler.

Reply:
removed